Kernel/Arch/x86/common/Interrupts.cpp

```diff
@@ -496,6 +498,23 @@ void handle_interrupt(TrapFrame* trap)
     clac();
     auto& regs = *trap->regs;
     VERIFY(regs.isr_number >= IRQ_VECTOR_BASE && regs.isr_number <= (IRQ_VECTOR_BASE + GENERIC_INTERRUPT_HANDLERS_COUNT));
+    // Note: we declare interrupt service routine offset 0x90 to 0x9f as
+    // reserved for when the PIC is disabled, so we can still route spurious
+    // IRQs to a different interrupt handlers at different location.
+    if (regs.isr_number >= pic_disabled_vector_base && regs.isr_number <= pic_disabled_vector_end) {
+        u8 irq = (u8)(regs.isr_number - pic_disabled_vector_base);
+        GenericInterruptHandler* handler = nullptr;
+        if (irq == 7) {
+            handler = s_disabled_interrupt_handler[0];
+        } else if (irq == 15) {
+            handler = s_disabled_interrupt_handler[1];
+        }
+        VERIFY(handler);
+        handler->increment_invoking_counter();
+        handler->handle_interrupt(regs);
+        handler->eoi();
+        return;
+    }
```

Comment:
I don't leave this (both the hardcoded 7/15 and the duplicated increment_invoking_counter/handle_interrupt/eoi, but as you said you intend to refactor this, I guess it's fine.

Reply:
I will keep a note for this when I refactor the low-level interrupt handling code.

Reply:
Fixed this now, I think it's fine until we do a refactor for this code :)